task_processor/decorators.py

```diff
@@ -0,0 +1,195 @@
+                    task_identifier=self.task_identifier,
+                    scheduled_for=delay_until or timezone.now(),
+                    priority=self.priority,
+                    queue_size=self.queue_size,
+                    args=args,
+                    kwargs=kwargs,
+                )
+            except TaskQueueFullError as e:
+                logger.warning(e)
+                return
+
+            task.save()
+            return task
+
+    def run_in_thread(
+        self,
+        *,
+        args: tuple[typing.Any] = (),
+        kwargs: dict[str, typing.Any] | None = None,
+    ) -> None:
+        _validate_inputs(*args, **kwargs)
+        thread = Thread(target=self.unwrapped, args=args, kwargs=kwargs, daemon=True)
+
+        def _start() -> None:
+            logger.info(
+                "Running function %s in unmanaged thread.", self.unwrapped.__name__
+            )
+            thread.start()
+
+        if self.transaction_on_commit:
+            return on_commit(_start)
+        return _start()
+
+
+def register_task_handler(  # noqa: C901
+    *,
+    task_name: str | None = None,
+    queue_size: int | None = None,
+    priority: TaskPriority = TaskPriority.NORMAL,
+    transaction_on_commit: bool = True,
+) -> typing.Callable[[typing.Callable[P, None]], TaskHandler[P]]:
+    """
+    Turn a function into an asynchronous task.
+
+    :param str task_name: task name. Defaults to function name.
+    :param int queue_size: (`TASK_PROCESSOR` task run method only)
+        max queue size for the task. Task runs exceeding the max size get dropped by
+        the task processor Defaults to `None` (infinite).
+    :param TaskPriority priority: task priority.
+    :param bool transaction_on_commit: (`SEPARATE_THREAD` task run method only)
+        Whether to wrap the task call in `transanction.on_commit`. Defaults to `True`.
+        We need this for the task to be able to access data committed with the current
+        transaction. If the task is invoked outside of a transaction, it will start
+        immediately.
+        Pass `False` if you want the task to start immediately regardless of current
+        transaction.
+    :rtype: TaskHandler
+    """
+
+    def wrapper(f: typing.Callable[P, None]) -> TaskHandler[P]:
+        return TaskHandler(
+            f,
+            task_name=task_name,
+            queue_size=queue_size,
+            priority=priority,
+            transaction_on_commit=transaction_on_commit,
+        )
+
+    return wrapper
+
+
+def register_recurring_task(
+    run_every: timedelta,
+    task_name: str | None = None,
+    args: tuple[typing.Any] = (),
+    kwargs: dict[str, typing.Any] | None = None,
+    first_run_time: time | None = None,
+) -> typing.Callable[[typing.Callable[..., None]], RecurringTask]:
+    if not os.environ.get("RUN_BY_PROCESSOR"):
+        # Do not register recurring tasks if not invoked by task processor
+        return lambda f: f
```

Comment:
Should `RUN_BY_PROCESSOR` be in `settings.py`?

Reply:
I have a feeling there's a reason that this is done (perhaps because settings are loaded at the time we need the env var) and we copied it from somewhere in the django project itself I believe. @gagantrivedi may remember further details?

Reply:
I'd argue that is not really a setting(in the global scope of things)
The use case, this:
```
       # The `register_recurring_task` decorator is executed in the global scope,
        # meaning it will run whenever the code is executed, including when running
        # `manage.py` commands such as `migrate` or `showmigrations`. However, this
        # decorator saves the task to the database, which is not desired unless it is
        # being run by processor, and to indicate the we set this `RUN_BY_PROCESSOR`
        # environment variable.
        
        ```